backend/prisma/db/ServiceCall.ts

```diff
@@ -83,15 +87,51 @@ export async function deleteServiceCall(id: number) {
 }
 
 // SEARCH
-export async function searchByString(search: string, page: number) {
+export async function search(search: SearchSchema) {
+  const where = [
+    // String search
+    { customerRequest: { contains: search.searchString } },
+    { employeeNotes: { contains: search.searchString } },
+    { notApprovedNotes: { contains: search.searchString } },
+
+    // Boolean search
+    { isApproved: search.searchBoolean },
+    { ATOOperational: search.searchBoolean },
+    { ATOReservoirFilled: search.searchBoolean },
+    { chemFilterAdjusted: search.searchBoolean },
+    { doserAdjustementOrManualDosing: search.searchBoolean },
+    { dosingReservoirsFull: search.searchBoolean },
+    { floorsCheckedForSpillsOrDirt: search.searchBoolean },
+    { glassCleanedInside: search.searchBoolean },
+    { glassCleanedOutside: search.searchBoolean },
+    { mechFilterChanged: search.searchBoolean },
+    { pumpsClearedOfDebris: search.searchBoolean },
+    { saltCreepCleaned: search.searchBoolean },
+    { skimmerCleanedAndOperational: search.searchBoolean },
+    { waterChanged: search.searchBoolean },
+    { waterTestedRecordedDated: search.searchBoolean },
+    { pestAPresent: search.searchBoolean },
+    { pestBPresent: search.searchBoolean },
+    { pestCPresent: search.searchBoolean },
+    { pestDPresent: search.searchBoolean },
+
+    // Number search
+    { alkalinity: { gte: search.minNum, lte: search.maxNum } },
+    { calcium: { gte: search.minNum, lte: search.maxNum } },
+    { nitrate: { gte: search.minNum, lte: search.maxNum } },
+    { phosphate: { gte: search.minNum, lte: search.maxNum } },
+
+    // Date search
+    { createdOn: { gte: search.minDate, lte: search.maxDate } },
+    { notesUpdated: { gte: search.minDate, lte: search.maxDate } },
+    // This removes any undefined values.
+  ].filter(Boolean);
```

Comment:
```suggestion
  ];
```
Correct me if I'm wrong, but the code runs the exact same with or without this filter. Prisma ignores undefined values